pallets/rmrk-core/src/lib.rs

```diff
@@ -690,3 +658,39 @@ pub mod pallet {
 		}
 	}
 }
+
+impl<T: Config> Collection<StringLimitOf<T>, T::AccountId> for Pallet<T> {
+	type CollectionId = T::CollectionId;
+
+	fn issuer(collection_id: Self::CollectionId) -> Option<T::AccountId> {
+		None
+	}
+	fn create_collection(
+		issuer: T::AccountId,
+		metadata: StringLimitOf<T>,
+		max: u32,
+		symbol: StringLimitOf<T>,
+		id: StringLimitOf<T>,
+	) -> sp_std::result::Result<Self::CollectionId, DispatchError> {
+		let collection = CollectionInfo { issuer: issuer.clone(), metadata, max, symbol, id };
+		let collection_id = <CollectionIndex<T>>::try_mutate(
+			|n| -> sp_std::result::Result<Self::CollectionId, DispatchError> {
+				let id = *n;
+				ensure!(id != Self::CollectionId::max_value(), Error::<T>::NoAvailableCollectionId);
+				*n += One::one();
+				Ok(id)
+			},
+		)?;
+		Collections::<T>::insert(collection_id, collection);
+		Ok(collection_id)
+	}
+
+	fn burn_collection(issuer: T::AccountId, collection_id: T::CollectionId) -> DispatchResult {
+		ensure!(
+			NFTs::<T>::iter_prefix_values(collection_id).count() == 0,
```

Comment:
Sounds like an O(n) scan. Maybe need to think about if we can avoid the iteration here

Reply:
Created separate issue